### PR TITLE
Rewrite median moe as a class and add loggin for clarity

### PR DESCRIPTION
**Pull request by SPTKL**

### CODE REVIEW

File: factfinder/calculate.py [hunk 1]
```diff
@@ -150,34 +151,18 @@ def calculate_e_m_median(self, pff_variable: str, geotype: str) -> pd.DataFrame:
...
-        # Empty dataframe to store the results
-        results = pd.DataFrame()
+        def get_median_and_median_moe(ranges, row, DF):
+            md = Median(ranges, row, DF)
+            e = md.median
+            m = md.median_moe
+            return pd.Series({'e': e, 'm': m})
+
+        results = df_pivoted.e.apply(lambda x: get_median_and_median_moe(
+            ranges, x, DF=design_factor), axis=1)
```
Comment (SPTKL) [inline factfinder/calculate.py:161]: @mgraber  
I did the same thing in the current branch. this is a lot easier

Comment (SPTKL) [reply factfinder/calculate.py:161]: #190

Comment (mgraber) [reply factfinder/calculate.py:161]: Ah, I missed this. Thanks!

**Review by mgraber — CHANGES_REQUESTED** [review]
This still produces errors when running. We get an error about min being calculated over an empty sequence when setting upper_bin. #192 might address this, but still has errors when run. I am trying to identify those more specifically now.

**Comment by mgraber** [comment]
Running [here](https://github.com/NYCPlanning/db-factfinder/runs/3221513551?check_suite_focus=true)

**Comment by SPTKL** [comment]
image
I archived the previous tables so that we can check the numbers

**Comment by mgraber** [comment]
@SPTKL  I believe the tables saved in _staging_Y2010-G2010_to_2020 are intermediate versions build from this branch, not ones built from dev.

**Comment by mgraber** [comment]
It looks as if there are fewer large discrepancies with the sample data, with no obvious systematic errors introduced. I think we can merge this branch then work on remaining edge cases #178.

**Pull request merged**